### ISSUE TITLE
align-assemblies error

### ISSUE THREAD

**Issue by beantkapoor786** [maize-genetics/phg_v2#144]
Hello, thank you for the version 2 of the PHG tool. I have six diploid genomes (12 haplotypes) of a fruit tree species. I am trying to align 11 haplotypes to a chosen reference haplotype however I am getting an error. The `align-assemblies` wrapper script runs for some time and then stops. I have run this command several times now and getting the same error. This command created `.maf` files for 4 of the haplotypes and then throws this error. 

Command - 
```bash
phg align-assemblies \
	--gff data/fortune_v1.4.0_primary.gff3 \
	--reference-file output/updated_assemblies/fortune_primary.fa \
	--assemblies data/assemblies.txt \
	--total-threads 30 \
	--in-parallel 4 \
	-o output/alignment_files/
```
Error - 
```bash
SLF4J: No SLF4J providers were found.
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See https://www.slf4j.org/codes.html#noProviders for further details.
SLF4J: Class path contains SLF4J bindings targeting slf4j-api versions 1.7.x or earlier.
SLF4J: Ignoring binding found at [jar:file:/pickett_flora/software/phg/lib/logback-classic-1.2.6.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See https://www.slf4j.org/codes.html#ignoredBindings for an explanation.
Exception in thread "main" java.nio.file.NoSuchFileException: /pickett_flora/projects/citrus_pangenome/analysis/4_phg_v2/lets-plot-images/output/alignment_files/australis_alternate_dotplot.svg
	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
	at java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:261)
	at java.base/java.nio.file.Files.newByteChannel(Files.java:379)
	at java.base/java.nio.file.Files.createFile(Files.java:657)
	at org.jetbrains.letsPlot.export.GgsaveKt.ggsave(ggsave.kt:73)
	at org.jetbrains.letsPlot.export.GgsaveKt.ggsave$default(ggsave.kt:44)
	at net.maizegenetics.phgv2.cli.AlignAssemblies.runAnchorwaveProali(AlignAssemblies.kt:616)
	at net.maizegenetics.phgv2.cli.AlignAssemblies.access$runAnchorwaveProali(AlignAssemblies.kt:69)
	at net.maizegenetics.phgv2.cli.AlignAssemblies$alignAssembly$2.invokeSuspend(AlignAssemblies.kt:508)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)
	Suppressed: java.nio.file.NoSuchFileException: /pickett_flora/projects/citrus_pangenome/analysis/4_phg_v2/lets-plot-images/output/alignment_files/australis_primary_dotplot.svg
		... 17 more
	Suppressed: java.nio.file.NoSuchFileException: /pickett_flora/projects/citrus_pangenome/analysis/4_phg_v2/lets-plot-images/output/alignment_files/australasica_primary_dotplot.svg
		... 17 more
	Suppressed: java.nio.file.NoSuchFileException: /pickett_flora/projects/citrus_pangenome/analysis/4_phg_v2/lets-plot-images/output/alignment_files/australasica_alternate_dotplot.svg
		... 17 more
```
Please let me know if you have suggestions. Thank you.

**Comment by lynnjo** [comment]
Beatrice – does this work if you give a full path to your output directory?

From: Beant Kapoor ***@***.***>
Date: Wednesday, April 10, 2024 at 2:43 PM
To: maize-genetics/phg_v2 ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [maize-genetics/phg_v2] align-assemblies error (Issue #144)

Hello, thank you for the version 2 of the PHG tool. I have six diploid genomes (12 haplotypes) of a fruit tree species. I am trying to align 11 haplotypes to a chosen reference haplotype however I am getting an error. The align-assemblies wrapper script runs for some time and then stops. I have run this command several times now and getting the same error. This command created .maf files for 4 of the haplotypes and then throws this error.

Command -

phg align-assemblies \

        --gff data/fortune_v1.4.0_primary.gff3 \

        --reference-file output/updated_assemblies/fortune_primary.fa \

        --assemblies data/assemblies.txt \

        --total-threads 30 \

        --in-parallel 4 \

        -o output/alignment_files/

Error -

SLF4J: No SLF4J providers were found.

SLF4J: Defaulting to no-operation (NOP) logger implementation

SLF4J: See https://www.slf4j.org/codes.html#noProviders for further details.

SLF4J: Class path contains SLF4J bindings targeting slf4j-api versions 1.7.x or earlier.

SLF4J: Ignoring binding found at [jar:file:/pickett_flora/software/phg/lib/logback-classic-1.2.6.jar!/org/slf4j/impl/StaticLoggerBinder.class]

SLF4J: See https://www.slf4j.org/codes.html#ignoredBindings for an explanation.

Exception in thread "main" java.nio.file.NoSuchFileException: /pickett_flora/projects/citrus_pangenome/analysis/4_phg_v2/lets-plot-images/output/alignment_files/australis_alternate_dotplot.svg

        at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)

        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106)

        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)

        at java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:261)

        at java.base/java.nio.file.Files.newByteChannel(Files.java:379)

        at java.base/java.nio.file.Files.createFile(Files.java:657)

        at org.jetbrains.letsPlot.export.GgsaveKt.ggsave(ggsave.kt:73)

        at org.jetbrains.letsPlot.export.GgsaveKt.ggsave$default(ggsave.kt:44)

        at net.maizegenetics.phgv2.cli.AlignAssemblies.runAnchorwaveProali(AlignAssemblies.kt:616)

        at net.maizegenetics.phgv2.cli.AlignAssemblies.access$runAnchorwaveProali(AlignAssemblies.kt:69)

        at net.maizegenetics.phgv2.cli.AlignAssemblies$alignAssembly$2.invokeSuspend(AlignAssemblies.kt:508)

        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)

        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)

        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)

        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)

        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)

        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)

        Suppressed: java.nio.file.NoSuchFileException: /pickett_flora/projects/citrus_pangenome/analysis/4_phg_v2/lets-plot-images/output/alignment_files/australis_primary_dotplot.svg

               ... 17 more

        Suppressed: java.nio.file.NoSuchFileException: /pickett_flora/projects/citrus_pangenome/analysis/4_phg_v2/lets-plot-images/output/alignment_files/australasica_primary_dotplot.svg

               ... 17 more

        Suppressed: java.nio.file.NoSuchFileException: /pickett_flora/projects/citrus_pangenome/analysis/4_phg_v2/lets-plot-images/output/alignment_files/australasica_alternate_dotplot.svg

               ... 17 more

Please let me know if you have suggestions. Thank you.

**Comment by lynnjo** [comment]
I have fixed this issue, will close this when the fix has been integrated to the build.

**Comment by beantkapoor786** [comment]
Thank you for the quick reply. I ran this command with the absolute path to the output directory yesterday afternoon and it finished without any error this morning. 

Thank you for the help. Cheers!

**Comment by lynnjo** [comment]
This has been fixed in release 2.2.77.126.  Relative paths are now properly processed by ggsave() (which writes the dotplot alignment to .svg file)
